Classes/common/CDTEncryptionKey/CDTEncryptionKeyNilProvider.m

```diff
@@ -0,0 +1,36 @@
+//
+//  CDTEncryptionKeyNilProvider.m
+//
+//
+//  Created by Enrique de la Torre Fernandez on 20/02/2015.
+//
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import "CDTEncryptionKeyNilProvider.h"
+
+@implementation CDTEncryptionKeyNilProvider
+
+#pragma mark - NSObject methods
+- (BOOL)isEqual:(id)object
+{
+    return (object && [object isMemberOfClass:[CDTEncryptionKeyNilProvider class]]);
+}
```

Comment:
The problem here is that I do not need `isEqual:`, as far as I can remember I overrided this method to perform some test that I discarded later on.